# Zebrafish pangenome project
We have devised two separate strategies for assembling the zebrafish genome “telomere-to-telomere” (T2T): The first strategy uses homozygous TU fish generated by heat shock disruption of mitosis I. Cell lines were established for two HS diploid fish and then genomic DNA was isolated both from the adult tissues as well as the cultured fibroblasts. DNA from the tissue was used for **PacBio HiFi** sequencing and the fibroblast DNA was used for **Oxford Nanopore sequencing (ONT)**. We performed *de novo* assembly using **Verkko** genome assembler. The second strategy we termed the **“3 generations”** approach. One fish from each of the 4 most commonly used zebrafish laboratory strains: **TU, AB, WIK, and TL** were used as the **4 “grandparents”** and were short read sequenced to document all unique, identifying structural nucleotide variants [SNV] for each parent. One fish from each of the 2 grandparent pairings was used to generate pools of the 3rd generation offspring. The pooled genomic DNA from these offspring was used for both **PacBio HiFi and ONT sequence**. The SNV data from the grandparents is then used to separate all the reads into haplotype bins and all 4 haplotypes are resolved simultaneously using the Verkko assembler.

## DNA extraction and sequencing
To be added

## Proposed analysis plan
### Genome assembly
- [x] The two widely used genome assembler; [Verkko](https://github.com/marbl/verkko)  will be used assemble the genome.  
- [x] Assembly quality assessments

### Gene annotation

### Methylation pattern analysis

### Repeat annotation

### Variant calling and analysis

